qiskit_machine_learning/algorithms/inference/qbayesian.py

```diff
@@ -0,0 +1,331 @@
+        """
+        # If evidence is empty
+        if len(evidence) == 0:
+            # Create circuit
+            qc = QuantumCircuit(*self._circ.qregs)
+            qc.append(self._circ, self._circ.qregs)
+            # Measure
+            qc.measure_all()
+            # Run circuit
+            self.samples = self._run_circuit(qc)
+            return self.samples
+        # Get Grover operator if evidence not empty
+        grover_op = self._get_grover_op(evidence)
+        # Amplitude amplification
+        true_e = {(self._label2qubit[e_key], e_val) for e_key, e_val in evidence.items()}
+        meas_e: Set[Tuple[str, int]] = set()
+        best_qc, best_inter = QuantumCircuit(), 0
+        self.converged = False
+        k = -1
+        # If the measurement of the evidence qubits matches the evidence stop
+        while (true_e != meas_e) and (k < self.limit):
+            # Increment power
+            k += 1
+            # Create circuit with 2^k times Grover operator
+            qc, meas_e = self.__power_grover(grover_op=grover_op, evidence=evidence, k=k)
+            # Test number of
+            if len(true_e.intersection(meas_e)) > best_inter:
+                best_qc = qc
+        if true_e == meas_e:
+            self.converged = True
+
+        # Create a classical register with the size of the evidence
+        best_qc_meas = QuantumCircuit(*self._circ.qregs)
+        best_qc_meas.append(best_qc, self._circ.qregs)
+        measurement_qcr = ClassicalRegister(self._circ.num_qubits - len(evidence))
+        best_qc_meas.add_register(measurement_qcr)
+        # Map the query qubits to the classical bits and measure them
+        query_qubits = [
+            (label, self._label2qidx[label], qubit)
+            for label, qubit in self._label2qubit.items()
+            if label not in evidence
+        ]
+        query_qubits_sorted = sorted(query_qubits, key=lambda x: x[1], reverse=True)
+        # Measure query variables and return their count
+        best_qc_meas.measure([q[2] for q in query_qubits_sorted], measurement_qcr)
+        # Run circuit
+        counts = self._run_circuit(best_qc_meas)
+        # Build default string with evidence
+        query_string = ""
+        var_idx_sorted = [
+            label for label, _ in sorted(self._label2qidx.items(), key=lambda x: x[1])
+        ]
+        for var in var_idx_sorted:
+            if var in evidence:
+                query_string += str(evidence[var])
+            else:
+                query_string += "q"
+        # Retrieve valid samples
+        self.samples = {}
+        # Replace placeholder q with query variables from samples
+        for key, val in counts.items():
+            query = query_string
+            for char in key:
+                query = query.replace("q", char, 1)
+            self.samples[query] = val
+        return self.samples
+
+    def inference(
+        self,
+        query: Dict[str, int],
+        evidence: Dict[str, int] = None,
```

Comment:
For places like this and the above with also returns a dict might it make sense to elaborate the key value pair meaning. I see `str` which can evidently take a bunch of different values. Maybe in this inference that is more free form if I skim over the tutorial but in other cases perhaps the strings has other meaning.

Reply:
Make sense. I will do so.

Reply:
I have now also added a method (_format_samples). This converts the samples from rejection_sampling back into their variable names and makes the result more accessible.